src/chatproviderservice/src/Client/ChatProviderServiceClient.lua

```diff
@@ -36,6 +36,20 @@ function ChatProviderServiceClient:Start()
 	TextChatService.OnIncomingMessage = function(textChatMessage)
 		self.MessageIncoming:Fire(textChatMessage)
 
+		local metadata = textChatMessage.Metadata
+		if metadata then
+			local isValidColor = pcall(function()
+				return Color3.fromHex(metadata)
```

Comment:
I would recommend JSONEncoding a table here instead and then still pcall everything, but that way we don't conflict if we send future metadata through here.

Reply:
resolved in [20071a8](https://github.com/Quenty/NevermoreEngine/pull/520/commits/20071a874fd7dbe733fdc116789fa8993ef0a405)